lib/concurrent/promise.rb

```diff
@@ -438,6 +438,46 @@ def self.any?(*promises)
       aggregate(:any?, *promises)
     end
 
+    # Aggregates a collection of promises, and completes as soon as the first
+    # aggregated promise completes, with the same state (fulfilled or rejected)
+    # and reason/value.
+    #
+    #   @param [Array] promises Zero or more promises to aggregate
+    #   @return [Promise] an unscheduled (not executed) promise that aggregates
+    #     the promises given as arguments
+    def self.race(*promises)
+      return Promise.fulfilled if promises.empty?
+
+      Promise.new do
+        latch = Concurrent::CountDownLatch.new(1)
+        finished = Concurrent::AtomicBoolean.new(false)
```

Comment:
If a latch count_down could return a different value depending on if it actually changed the value of the latch or did not, that would be all that was neccesary.  The only time `count_down` does not change the value is if it was already 0, of course. Or, I suppose, if an AtomicBoolean had a way to 'wait for it to become true', then it could serve the purpose instead of the latch.  I don't _think_ either of those are presently true?